src/Core/Vault/Repositories/IFolderRepository.cs

```diff
@@ -9,6 +9,10 @@ public interface IFolderRepository : IRepository<Folder, Guid>
     Task<Folder> GetByIdAsync(Guid id, Guid userId);
     Task<ICollection<Folder>> GetManyByUserIdAsync(Guid userId);
 
+    [Obsolete("Not used while FlexibleCollections feature flag is in use - use the overload to pass in the flag value instead.")]
+    new Task DeleteAsync(Folder folder);
+    Task DeleteAsync(Folder folder, bool useFlexibleCollections);
```

Comment:
This is a bit awkward, but the base repository class provides an implementation of `DeleteAsync`, which we want to prevent use of (because you need to use the overload to pass in the flag value).